File: Common/Entity/src/main/java/it/ey/dto/Sezione31DTO.java
package it.ey.dto;

import lombok.*;
import lombok.experimental.SuperBuilder;

import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@SuperBuilder(toBuilder = true)
public class Sezione31DTO extends SezioneBaseDTO
{
    private Long id;

    private Long idPiao;

    private String statoSezione;

    private String strutturaOrganizzativaAP;

    //commento
    private String ampiezzaOrganica;

    private String incarichiDirigenziali;

    private String profiliProfessionali;

    private String lineeOrganizzazione;

    private List<AmpiezzaOrganizzativaDTO> ampiezzaOrganizzative;

    private List<TabellaFunzionaleDTO> tabelleFunzionali;

    private Boolean graficoMinerva;
}

File: dfp-piao-fe/src/app/shared/models/classes/sezione-31-dto.ts
import { AllegatoDTO } from './allegato-dto';
import { AmpiezzaOrganizzativaDTO } from './ampiezza-organizzativa-dto';
import { CampiTecniciDTO } from './campi-tecnici-dto';
import { TabellaFunzionaleDTO } from './tabella-funzionale-dto';
import { UlterioriInfoDTO } from './ulteriori-info-dto';

export class Sezione31DTO extends CampiTecniciDTO {
  id!: number;
  idPiao!: number;
  statoSezione!: string;
  // Struttura organizzativa al 31.12 dell'anno precedente
  strutturaOrganizzativaAP!: string;
  // Ampiezza organica
  ampiezzaOrganica!: string;
  // Incarichi dirigenziali e simili
  incarichiDirigenziali!: string;
  // Rappresentazione dei profili professionali
  profiliProfessionali!: string;
  // Linee organizzazione
  lineeOrganizzazione!: string;

  ampiezzaOrganizzative?: AmpiezzaOrganizzativaDTO[];

  tabelleFunzionali?: TabellaFunzionaleDTO[];

  ulterioriInfo?: UlterioriInfoDTO;

  //Allegati
  allegati?: AllegatoDTO[];

  campiModificati?: string;
  testoSezione?: string;

  // NUOVO FLAG
  graficoMinerva?: boolean;
}
